### PR TITLE
Genprint 0.1

### PR DESCRIPTION
**Pull request by progman1**
library for printing of values a la toplevel but within compilation units.

### CODE REVIEW

**Comment by progman1** [comment]
this is installing on my machine with 'opam install .'
but is failing in CI on 'dune build -p genprint' with
# Error: I don't know about package genprint (passed through --only-packages/--release)

anyone know what's going on?

File: packages/Genprint/Genprint.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+description: """
+A PPX syntax extension and library package enabling printing of values of any type using OCaml's internal printing facilities a la toplevel evaluation.
+Useful for debugging as a quick alternative to ocamldebug/ppx_deriving/#install_printer.
+"""
+
+maintainer: "progman1"
+authors: [ "progman1" ]
+homepage: "github.com/progman1/genprintlib"
+dev-repo: "git+https://github.com/progman1/genprintlib.git"
+bug-reports: "progman1"
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]:
```suggestion
bug-reports: "https://github.com/progman1/genprintlib/issues"
```

File: packages/Genprint/Genprint.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+version: "0.1"
+synopsis: "PPX syntax extension and library package for printing values of any type"
+description: """
+A PPX syntax extension and library package enabling printing of values of any type using OCaml's internal printing facilities a la toplevel evaluation.
+Useful for debugging as a quick alternative to ocamldebug/ppx_deriving/#install_printer.
+"""
+
+maintainer: "progman1"
+authors: [ "progman1" ]
+homepage: "github.com/progman1/genprintlib"
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]:
```suggestion
homepage: "https://github.com/progman1/genprintlib"
```

File: packages/Genprint/Genprint.0.1/opam [hunk 3]
```diff
@@ -0,0 +1,24 @@
+opam-version: "2.0"
+name: "Genprint"
+version: "0.1"
+synopsis: "PPX syntax extension and library package for printing values of any type"
+description: """
+A PPX syntax extension and library package enabling printing of values of any type using OCaml's internal printing facilities a la toplevel evaluation.
+Useful for debugging as a quick alternative to ocamldebug/ppx_deriving/#install_printer.
+"""
+
+maintainer: "progman1"
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]: would it be possible to add an email address?

File: packages/Genprint/Genprint.0.1/opam [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+homepage: "github.com/progman1/genprintlib"
+dev-repo: "git+https://github.com/progman1/genprintlib.git"
+bug-reports: "progman1"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" "genprint"]
+]
+
+depends: [
+  "ocaml" {build} # {>= "4.07.1"}
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]:
```suggestion
  "ocaml" {>= "4.07.1"}
```

File: packages/Genprint/Genprint.0.1/opam [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+"""
+
+maintainer: "progman1"
+authors: [ "progman1" ]
+homepage: "github.com/progman1/genprintlib"
+dev-repo: "git+https://github.com/progman1/genprintlib.git"
+bug-reports: "progman1"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" "genprint"]
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]:
```suggestion
  ["dune" "build" "-p" "genprint" "-j" jobs]
```
This adds people to choose how many cores they want to give to compile your package

**Comment by kit-ty-kate** [comment]
> anyone know what's going on?

Yes, your package doesn't define where to get the url from (see the url section in other packages for example)

File: packages/Genprint/Genprint.0.1/opam [hunk 6]
```diff
@@ -0,0 +1,24 @@
+opam-version: "2.0"
+name: "Genprint"
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]: This field is redundant. Also, could you avoid capitalized names?

File: packages/Genprint/Genprint.0.1/opam [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+version: "0.1"
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]: This field is redundant

File: packages/Genprint/Genprint.0.1/opam [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+authors: [ "progman1" ]
+homepage: "github.com/progman1/genprintlib"
+dev-repo: "git+https://github.com/progman1/genprintlib.git"
+bug-reports: "progman1"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" "genprint"]
+]
+
+depends: [
```
Comment (kit-ty-kate) [inline packages/Genprint/Genprint.0.1/opam]: Your project requires dune to build but is not defined in the dependencies

**Pull request closed without merging**